### ISSUE TITLE
ConeGeometry contains degenerate triangles

### ISSUE THREAD

**Issue by gkjohnson** [mrdoob/three.js#29454]
### Description

ConeGeometry extends CylinderGeometry and simply [sets the top radius to 0](https://github.com/mrdoob/three.js/blob/dev/src/geometries/ConeGeometry.js#L7) resulting in one triangle of all the 2-triangle rectanglar sides being reduced to an edge. Eg a cone with 3 sides should require a total of 6 triangles - but instead consists of 9. 3 of which are degenerate triangles.

<img width="240" alt="image" src="https://github.com/user-attachments/assets/7b1e7695-3579-4a85-a0fd-d7d3101794a4">

You can see the triangles that are being reduced to an edge as the top radius gets smaller:

<img width="237" alt="image" src="https://github.com/user-attachments/assets/c2c13a60-98a5-4405-81fa-878bc8bb4afc">

Degenerate triangles like this can cause issues with any algorithms that require computing triangle normals, take surface area, generating half-edge structures, etc. It would be great if the core three.js primitives could be properly constructed without degenerate triangle issues.

### Reproduction steps

1. Create a cone geometry
2. Check the number of triangles
3. There are are more triangles than necessary

### Code

```js
const cone = new THREE.ConeGeometry( 1, 1, 3, 1 );
console.log( cone.index.count / 3 ); // logs 9 instead of 6
```

### Live example

* [jsfiddle-latest-release WebGLRenderer](https://jsfiddle.net/3mrkqyea/)
* [jsfiddle-dev WebGLRenderer](https://jsfiddle.net/gcqx26jv/)
* [jsfiddle-latest-release WebGPURenderer](https://jsfiddle.net/8L2jkmx7/)
* [jsfiddle-dev WebGPURenderer](https://jsfiddle.net/L3n1w4yh/)

### Version

dev

**Comment by andrewhodel** [comment]
It's probably using a double instead of an integer with a known range.

**Comment by Mugen87** [comment]
Since the issue is located in the torso generation of `CylinderGeometry`, how about we change:

https://github.com/mrdoob/three.js/blob/a58d0e9224f74cf50a13aecd1e4e7461e95014c6/src/geometries/CylinderGeometry.js#L82

to

```js
let radius = v * ( radiusBottom - radiusTop ) + radiusTop; 
radius = ( radius === 0 ) ? Number.EPSILON : radius; // FIX
```
So with a zero radius, we use `Number.EPSILON ` to avoid degenerate triangles. I guess this would solve the issue, right?

If we handle the zero radius with a separate code path, I'm afraid the vertex and index generation will be more complex. Having three more triangles around should be an issue.

**Comment by gkjohnson** [comment]
Creating unnecessary almost-degenerate triangles and a slightly open-cap cone isn't a lot better imo. But we should just be able to skip adding the degenerate triangles if we're okay with modifying CylinderGeometry. Fixed in #29460
